### PR TITLE
Docs Clean update

### LINKED ISSUE #37: filter_entries doesn't take an int status_code
According to the docs on https://pythonhosted.org/haralyzer/haralyzer.html filter_entries takes in a int as status_code.
However, when doing this, the regex search crashes, as the filter function actually appears to intend to take a string.

If I'm interpreting this situation correctly, the fix to this is simply to change the documentation to clarify that status_code should be passed as a string. Let me know if I'm reading it wrong though!

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Moves all the documentation away from a big readme to be more organized.

- Fixes #37 
- Improves #32 
- Fixed bug with building on the master branch

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,377 +14,11 @@ Haralyzer
...
-    single_entry.response.redirectURL -> String of the redirect URL or None
-    single_entry.response.status -> Integer of th HTTP status code
-    single_entry.response.statusText -> String of HTTP status
-    single_entry.response.text -> String of content received
-
-    ** You are still able to access items like a dictionary.
-    print(single_entry["connection"])
-    # "80"
+.. image:: https://img.shields.io/pypi/dm/haralyzer
+    :alt: PyPI - Downloads
```
Comment (Cyb3r-Jak3) [inline README.rst:18]: @mrname The REAME.md now looks really empty. Not sure if want anything filled in here,

File: .github/workflows/create_release.yml [hunk 2]
```diff
@@ -2,14 +2,13 @@ name: Create Release
...
   push:
-    branches:
-      - "master"
     tags:
       - 'v*'
 
 jobs:
   release:
     name: Create Release Asset
+    if: github.event.base_ref == 'refs/heads/master'
```
Comment (Cyb3r-Jak3) [inline .github/workflows/create_release.yml:11]: Makes it so the job only runs if the master branch and tagged.

File: README.rst (hunk 1 above)
Comment (mrname) [reply README.rst:18]: Good call @Cyb3r-Jak3 . Could we link to the new nested pages? At some point we had readthedocs hooked up to there, so if we link down it should compile and upload to there (if that is still working).

**Review by mrname — APPROVED** [review]
Great work! Aside from handling the README this is a great idea to break things up.

File: README.rst (hunk 1 above)
Comment (Cyb3r-Jak3) [reply README.rst:18]: From my testing, the documentation is built from the docs folder so that is where the links really need to be.

**Comment by Cyb3r-Jak3** [comment]
@mrname Let me know what you think of the ReadMe now. I copied some of the content from the docs there and added a blurb about contributing. Looks a lot less empty now. If you think it is good then please merge. Not planning on a release for this PR as all documentation updates.

**Pull request merged**